### ISSUE TITLE
the output data after train,  why is all verified data 0, please help me

### ISSUE THREAD

**Issue by CarolYeh** [WongKinYiu/yolov9#614]
![exp4_results](https://github.com/user-attachments/assets/d9734663-dcac-40c5-a311-e52b2964b318)
![Image_1117_1712_002](https://github.com/user-attachments/assets/b798cbf1-af6d-4bc5-9cf2-0e12413ade61)
the red box is valid data, the data is 0, the top picture is the output report,

![Image_1117_1723_006](https://github.com/user-attachments/assets/76f10e23-13a4-4efb-bf15-064c864a8d2f)
the picture is  my YOLO run param  
I don't know what  happened, why val data is 0, please help me thank your

**Comment by ma1688** [comment]
` # Loss
  if compute_loss:
      preds = preds[1]
      #train_out = train_out[1]
      loss += compute_loss(train_out, targets)[1]  # box, obj, cls
  else:
      preds = preds[0][1]
`

replace val_trul.py

**Comment by Humorousf** [comment]
> `# Loss if compute_loss: preds = preds[1] #train_out = train_out[1] loss += compute_loss(train_out, targets)[1] # box, obj, cls else: preds = preds[0][1]`
> 
> 替换 val_trul.py

do you mean？
        # Loss
        if compute_loss:
            preds = preds[1]
            # train_out = train_out[1]
            loss += compute_loss(train_out, targets)[1]  # box, obj, cls
        else:
            preds = preds[0][1]

**Comment by fazrigading** [comment]
> > `# Loss if compute_loss: preds = preds[1] #train_out = train_out[1] loss += compute_loss(train_out, targets)[1] # box, obj, cls else: preds = preds[0][1]`
> > 替换 val_trul.py
> 
> do you mean？ # Loss if compute_loss: preds = preds[1] # train_out = train_out[1] loss += compute_loss(train_out, targets)[1] # box, obj, cls else: preds = preds[0][1]

Yes, uncomment the `loss = ...` line
